### PR TITLE
transport capture: rename to tap

### PR DESCRIPTION
**Pull request by mattklein123**
This is a rename PR only. It renames the capture transport socket
and associated tools to the tap transport socket. It also updates
some documentation. In a subsequent PR I'm going to refactor the
tap transport socket to use the new common tap framework so that
the tap transport socket can be configured via admin, the HTTP
tap filter can write to a file, the tap transport socket can have
matching, etc.

*Risk Level*: Low
*Testing*: Existing tests
*Docs Changes*: Done
*Release Notes*: I will add a complete release note when I do the 
full refactor in a forthcoming PR.

### CODE REVIEW

**Comment by mattklein123** [comment]
@htuch I think I got everything but let me know if you see anything missing. I didn't rename the "capture" fuzz tests to "tap" fuzz tests since I don't think it matters, but let me know if you prefer that and I can do that also.

File: ci/do_ci.sh [hunk 1]
```diff
@@ -111,19 +111,19 @@ elif [[ "$1" == "bazel.asan" ]]; then
...
-  CAPTURE_TMP=/tmp/capture/
-  rm -rf "${CAPTURE_TMP}"
-  mkdir -p "${CAPTURE_TMP}"
+  TAP_TMP=/tmp/tap/
+  rm -rf "${TAP_TMP}"
+  mkdir -p "${TAP_TMP}"
   bazel_with_collection test ${BAZEL_TEST_OPTIONS} -c dbg --config=clang-asan \
     @envoy//test/integration:ssl_integration_test \
-    --test_env=CAPTURE_PATH="${CAPTURE_TMP}/capture"
+    --test_env=TAP_PATH="${TAP_TMP}"
```
Comment (mattklein123) [inline ci/do_ci.sh]: @htuch I'm pretty sure this line was wrong (the extra `/capture`) and I don't think the below `ls -l` was actually checking for any output. Any bash suggestions on the best way to check to see if there is > 1 file below?

Comment (htuch) [reply ci/do_ci.sh]: I think the `capture` is the prefix of the filename, so was correct.

For the `ls`, this was verifying, try `ls -l *.foo || echo OK` in your shell if you don't believe me :)

File: api/envoy/data/tap/v2alpha/transport.proto [hunk 2]
```diff
@@ -47,7 +47,7 @@ message Event {
   }
 }
 
-// Sequence of read/write events that constitute a captured trace on a socket.
+// Sequence of read/write events that constitute a trace on a socket.
 // Multiple Trace messages might be emitted for a given connection ID, with the
 // sink (e.g. file set, network) responsible for later reassembly.
 message Trace {
```
Comment (htuch) [inline api/envoy/data/tap/v2alpha/transport.proto]: You're probably going to want to move from a "entire connection trace" to a chunked trace for streaming in a later PR.

File: ci/do_ci.sh (hunk 1 above)
Comment (mattklein123) [reply ci/do_ci.sh]: Ah OK. Let me take a look.

File: api/envoy/data/tap/v2alpha/transport.proto (hunk 2 above)
Comment (mattklein123) [reply api/envoy/data/tap/v2alpha/transport.proto]: Yeah I'm going to be working on the streaming stuff in later PRs for sure.

**Pull request merged**